coral/src/app/layout/header/HeaderMenuLink.test.tsx

```diff
@@ -1,61 +1,101 @@
+import data from "@aivenio/aquarium/dist/src/icons/console";
+import { cleanup, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import HeaderMenuLink from "src/app/layout/header/HeaderMenuLink";
-import { cleanup, screen, render, within } from "@testing-library/react";
-import data from "@aivenio/design-system/dist/src/icons/console";
+import { tabNavigateTo } from "src/services/test-utils/tabbing";
 
 // mock out svgs to avoid clutter
-jest.mock("@aivenio/design-system", () => {
+jest.mock("@aivenio/aquarium", () => {
   return {
     __esModule: true,
-    ...jest.requireActual("@aivenio/design-system"),
+    ...jest.requireActual("@aivenio/aquarium"),
 
     Icon: () => {
       return <div data-testid={"ds-icon"}></div>;
     },
   };
 });
 
+const linkText = "Go to your profile page";
+
 describe("HeaderMenuLink.tsx", () => {
   // (icon is not needed for the test, Icon component mocked out)
   const mockIcon = "" as unknown as typeof data;
 
   describe("renders a default link with required props", () => {
-    beforeAll(() => {
+    beforeEach(() => {
       render(
         <HeaderMenuLink
           icon={mockIcon}
           href={"/myProfile"}
-          linkText={"Go to your profile page"}
+          linkText={linkText}
         />
       );
     });
 
-    afterAll(cleanup);
+    afterEach(cleanup);
 
     it(`renders a link with text dependent on a given property`, () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
 
       expect(navLink).toBeVisible();
     });
 
-    it(`renders a href for that link dependent on a given  property`, () => {
+    it(`renders a href for that link dependent on a given property`, () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
       expect(navLink).toHaveAttribute("href", "/myProfile");
     });
 
-    it(`renders a Tooltip with an Icon, both hidden for assistive technology`, () => {
+    it(`renders an Icon`, async () => {
+      const icon = screen.getByTestId("ds-icon");
+
+      expect(icon).toBeVisible();
+    });
+
+    it(`triggers the rendering of a Tooltip on mouse hover`, async () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
-      const tooltip = within(navLink).getByTestId("tooltip");
-      const icon = within(tooltip).getByTestId("ds-icon");
+      const queryForToolTip = () => screen.queryByRole("tooltip");
```

Comment:
Even MORE nitpicky, I think we should not use the role here. Usually, using role is always the best way to go, but in this case, as far as I understand, the role: 

- is not really used as a ROLE in sense of "adds more context / information" because it's not usable by assistive technology
- it's not a proper `tooltip` in our case

MDN writes about role tooltip: 
> The tooltip is not the appropriate role for the more information "i" icon, ⓘ. A tooltip is directly associated with the owning element. The ⓘ isn't 'described by' detailed information; the tool or control is.

So using the `byRole` in the test for me would imply that we're testing for a proper `tooltip`, while the `<Tooltip />` in the code is aria-hidden, which should make that test impossible. I feel that can create confusion. 

I would wrap a `div` of `span` with aria-hidden around the `<Tooltip />`, that way our tests have to match the reality for users of assistive technologies and there's no discrepancy between code and test.

Reply:
I would argue that this does not apply in this case 🤔 
> The tooltip is not the appropriate role for the more information "i" icon, ⓘ. A tooltip is directly associated with the owning element. The ⓘ isn't 'described by' detailed information; the tool or control is.

Our icon links (the "tool or control") are "described by" the tooltip :o Every other point of the description before this point seems to also describe precisely our usecase (except for the delay, we don't have that currently).

This probably should be also relayed to the DS team so they can implement it in the `Tooltip`, though! Then we can dispense with the hidden span, probably :o

> The element that triggers the tooltip references the tooltip element with [aria-describedby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby).

Reply:
Ugh, I expected this to work but no. The `id` is properly forwarded to the `div` rendered with the `tooltipText`, but voice over does not pick up the text inside the div.

```
      <Tooltip
        // aria-hidden is currently not forwarded, but it is added for semantics purposes
        aria-hidden="true"
        content={linkText}
        placement="right"
        isOpen={isOpen}
        id={linkText}
      >
        {/*@TODO add correct link*/}
        {/* aria-hidden="true" is added natively to the Icon component */}
        <Icon
          icon={icon}
          fontSize={"20px"}
          color={"grey-0"}
          aria-describedby={linkText}
        />
      </Tooltip>
 ```

Reply:
Still added it, it seems correct [refactor(coral): Add aria-describedby](https://github.com/aiven/klaw/pull/329/commits/d1f52a971a034c1cd01e8165825d32c2ac4ab24f)

Reply:
>Our icon links (the "tool or control") are "described by" the tooltip :o Every other point of the description before this point seems to also describe precisely our usecase (except for the delay, we don't have that currently).
Hm. OK that's a very good point you're making now that I read it again 🙈  
All examples I found about accessbile tooltips had longer text where one word had a tooltip, I think that's why I landed there. 

>And also, maybe they are not picked up by the macos screenreader? Or maybe their behaviour is not what I assume?
I think the problem is that the element is added dynamically to the DOM element? I would have to read up more about that, accessible tooltips are not super easy :D  I found a few articles and WCAG requirements, but I think that's something that should go into the design system. 

Your implementation right now works as we expect, so, good for us!